src/lib/mongoBackend/mongoUpdateContextSubscription.cpp

```diff
@@ -266,5 +267,17 @@ HttpStatusCode mongoUpdateContextSubscription
   responseP->subscribeResponse.subscriptionId = requestP->subscriptionId;
 
   reqSemGive(__FUNCTION__, "ngsi10 update subscription request", reqSemTaken);
+
+
+  //
+  // Update the entry in the cache of subscriptions, if applicable
+  //
+  Subscription* subP = subCache->lookupById(tenant, servicePathV[0], requestP->subscriptionId.get());
```

Comment:
In this case, what about the ispattern and ONCHANGE checking? A ONCHANGE subscritpion could change to ONTIMEINTERVAL or viceversa. A subscription using isPattern=true could change to use isPattern=false (not legal in NGSIv1 put probably legal in NGSIv2) in a short future...

Thus, implementing update as remove + insert and do the checking in insert could be better strategy.